packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -424,11 +579,65 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
     properties.add(DiagnosticsProperty<Object>('owner', owner, level: parent != null ? DiagnosticLevel.hidden : DiagnosticLevel.info, defaultValue: null));
-    properties.add(DiagnosticsProperty<dynamic>('creator', debugCreator, defaultValue: null, level: DiagnosticLevel.debug));
-    properties.add(DiagnosticsProperty<String>('engine layer', describeIdentity(_engineLayer)));
+    properties.add(DiagnosticsProperty<Object?>('creator', debugCreator, defaultValue: null, level: DiagnosticLevel.debug));
+    if (_engineLayer != null) {
+      properties.add(DiagnosticsProperty<String>('engine layer', describeIdentity(_engineLayer)));
+    }
+    properties.add(DiagnosticsProperty<int>('handles', debugHandleCount));
   }
 }
 
+/// A handle to prevent a [Layer]'s platform graphics resources from being
+/// disposed.
+///
+/// [RenderObject]s that need to hold more than one [Layer] may use the
+/// [LayerHolder] class to help manage the lifetime of additional layers.
+///
+/// [Layer] objects retain native resourses such as [EngineLayer]s and [Picture]
+/// objects. These objects may in turn retain large chunks of texture memory,
+/// either directly or indirectly.
+///
+/// The layer's native resources must be retained as long as there is some
+/// object that can add it to a scene. Typically, this is either its
+/// [Layer.parent] or an undisposed [RenderObject] that will append it to a
+/// [ContainerLayer]. Layers automatically hold a handle to their children, and
+/// RenderObjects automatically hold a handle to their [RenderObject.layer] as
+/// well as any [PictureLayer]s that they paint into using the
+/// [PaintingContext.canvas]. A layer automatically releases its resources once
+/// at least one handle has been acquired and all handles have been disposed.
+/// [RenderObject]s that create additional layer objects must manually manage
+/// the handles for that layer similarly to the implementation of
+/// [RenderObject.layer].
+///
+/// Layers can be reused even after their native resources has been released,
+/// since they are mutable objects and the native resources can be reset on
+/// subsequent frames.
+class LayerHandle {
+  LayerHandle._(Layer this._layer);
+
+  Layer? _layer;
+
+  /// Whether the layer this object refers to is identical to `layer`.
+  ///
+  /// In debug mode, this will throw after [dispose] has been called.
+  bool isHandleForLayer(Layer layer) {
+    assert(_layer != null, 'Cannot check a disposed handle.');
+    return identical(layer, _layer);
+  }
+
+  /// Releases the reference to the [Layer] that created this handle.
+  ///
+  /// This object is unusable after this call.
+  void dispose() {
+    assert(_layer != null, 'Handle already disposed.');
+    _layer!._unref();
+    _layer = null;
+  }
+
+  @override
+  String toString() => 'LayerHandle(${_layer != null ? _layer.toString() : 'DISPOSED'})';
+}
+
 /// A composited layer containing a [Picture].
 ///
 /// Picture layers are always leaves in the layer tree.
```

Comment:
Since this class is responsible for calling `Picture.dispose` it is effectively the owner of the `Picture` object it holds. We should note that in the dartdocs.

Reply:
Added